### ISSUE TITLE
Change behaviour of ruleset over-ride for ID difficulty

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#657]
The approach currently taken is that if a given iRecord user account has had a species record accepted at least once, then that species is no longer flagged as being hard to ID for that user. The user will no longer receive notifications, and records of that species from that user will no longer get flagged to the verifier as having failed the ID rule. This is documented in the code, at line 65 here:
https://github.com/Indicia-Team/warehouse/blob/master/modules/data_cleaner_identification_difficulty/plugins/data_cleaner_identification_difficulty.php

This has been identified as a problem, since verifiers are likely to want to check **_all_** records of hard-to-identify species, regardless of whether the user has recorded them before, and if applying bulk verification procedures they will not necessarily want hard-to-identify species to be bulk verified. 

The proposal is for this to be split so that once a recorder has had a record accepted for the species in question they will no longer receive notifications about that species, but all records of the species will still be flagged up for verifiers.

The solution therefore is to make the identification_difficulty rule violation fire in all cases, but move the filtering out of records where the recorder has successfully recorded something into the notification system instead. That way the flag will exist on the record, but the recorder won't get any repeat notifications.

Originally discussed under #507

**Comment by johnvanbreda** [comment]
@burkmarr I had a play with a few alternatives (e.g. a not exists query) but found that your approach is the fastest and it also gave the correct results. When run against the last day's data (which is more realistic than your fairly large timespan) the increase in time was from 20ms to about 100ms so I am happy with that.

**Comment by burkmarr** [comment]
Pull request against devel generated.

**Comment by burkmarr** [comment]
Released in https://github.com/Indicia-Team/warehouse/releases/tag/v2.34.0.

**Comment by kitenetter** [comment]
In my verification grid (new page) I'm still seeing species that are flagged as 'hard to ID' for some recorders, and not flagged at all for others.

Is that because we have not succeeded in changing the rule behavious as described above?

Or is it a result of the change in behaviour not being applied retrospectively?

If it's the latter I can live with that, just wanted to check that the change had been successful.

**Comment by burkmarr** [comment]
Yes I think that it's because they've already been through the record cleaner under the old configuration and therefore the relevant occurrence comment record was not generated. You'll need to check with some records entered after the software update: https://github.com/Indicia-Team/warehouse/releases/tag/v2.34.0

**Comment by johnvanbreda** [comment]
@kitenetter are you happy to close this?

**Comment by kitenetter** [comment]
Seems to be working as expected from the verifiers' perspective. Closing.

**Comment by kitenetter** [comment]
This seems to be giving inconsistent results again - these are all for the same species, and all for records entered in July or August 2019:

![image](https://user-images.githubusercontent.com/16039111/72668306-543d7d80-3a1d-11ea-8cd3-34992ed269a2.png)

**Comment by burkmarr** [comment]
@kitenetter - the software fix wasn't released on live until September 30th.

**Comment by kitenetter** [comment]
Thanks - it feels like longer ago than that! Closing.

**Comment by kitenetter** [comment]
@burkmarr we've had another report of inconsistent flagging of ID difficulty, this time for butterflies, and including records added recently. The examples I've been given are all for Brown Argus, so maybe this is a taxon-specific anomaly.

Examples:
Occurrence ID 21766839, Brown Argus, passes all rule checks
Occurrence ID 21590129, Brown Argus, passes all rule checks
Occurrence ID 21759110, Brown Argus, flagged as hard to ID
Occurrence ID 21711079, Brown Argus, flagged as hard to ID

**Comment by kitenetter** [comment]
No problems reported recently. Closing.
